src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -22,13 +22,15 @@ public sealed class AspNetCoreMvc2Integration : IDisposable
         /// <summary>
         /// Type for unobtrusive hooking into Microsoft.AspNetCore.Mvc.Core pipeline.
         /// </summary>
-        private const string DiagnosticSource = "Microsoft.AspNetCore.Mvc.Internal.MvcCoreDiagnosticSourceExtensions";
+        private const string DiagnosticSourceTypeName = "Microsoft.AspNetCore.Mvc.Internal.MvcCoreDiagnosticSourceExtensions";
 
         /// <summary>
         /// Base type used for traversing the pipeline in Microsoft.AspNetCore.Mvc.Core.
         /// </summary>
-        private const string ResourceInvoker = "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker";
+        private const string ResourceInvokerTypeName = "Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker";
 
+        private static readonly Type DiagnosticSourceType = Type.GetType($"{DiagnosticSourceTypeName}, {AspnetMvcCore}");
```

Comment:
I don't see why not. If we call it multiple times in the same AppDomain it will return the same value, right?

Reply:
The first object in the `Rethrow` wrapper method is an instance of `Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker`. I think Colin is concerned about different versions of that type being loaded into the same AppDomain so if we just grab the type by using the object instance (instead of statically) we could eliminate that issue.

Reply:
I didn't realize before that these are static classes that hold the extension methods. I'm good with the way it is.